src/common/Loader.vue

```diff
@@ -0,0 +1,31 @@
+<template>
+  <div class="loader">
+    <template v-if="scheme === 'spinner'">
+      <spinner />
+    </template>
+    <template v-else-if="scheme === 'skeleton'">
+      <skeleton v-bind="$attrs" />
+    </template>
+  </div>
+</template>
+
+<script lang="ts" setup>
+import { Spinner, Skeleton } from '@/common/loaders'
+
+withDefaults(
+  defineProps<{
+    scheme: 'spinner' | 'skeleton'
+  }>(),
+  {
+    scheme: 'spinner',
+  },
+)
+</script>
+
+<style lang="scss" scoped>
+.loader {
+  overflow: hidden;
+  display: grid;
+  place-items: center;
+}
+</style>
```

Comment:
To improve code clarity move defining loader component to **script** section.

```ts
const loaderComponent = computed(() => {
  switch (props.scheme) {
    case 'skeleton':
      return Skeleton
    case 'spinner':
    default:
      return Spinner
  }
})
```
in **template**
```vue
<template>
  <div class="loader">
    <component v-bind="$attrs" :is="loaderComponent" />
  </div>
</template>
```

Reply:
done